src/charm.py

```diff
@@ -18,79 +18,56 @@
 )
 
 
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
+
+    def __init__(self, msg, status=None):
+        super().__init__()
+
+        self.msg = msg
+        self.status = status
```

Comment:
What do you think to us making a helpers library for things like this that are common across our codebase?  That way it'll be easier to be consistent.  There's other patterns we repeat across our repos as well that would be good to share

Reply:
I wouldn't mind, but I would look into it after the release.

Reply:
Minor change to remove boilerplate whenever we use this to set a status

```suggestion
class CheckFailed(Exception):
    """ Raise this exception if one of the checks in main fails. """

    def __init__(self, msg, status_type=None):
        super().__init__()

        self.msg = msg
        self.status_type = status_type
        self.status = status_type(msg)
```